app/src/main/java/org/wikipedia/search/RecentSearchesFragment.kt

```diff
@@ -155,4 +181,34 @@ class RecentSearchesFragment : Fragment() {
             holder.bindItem(pos)
         }
     }
+
+    private inner class NamespaceItemViewHolder constructor(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
+        private var ns: Namespace? = null
+
+        fun bindItem(ns: Namespace?) {
+            itemView.setOnClickListener(this)
+            this.ns = ns
+            (itemView as TextView).text = if (ns == null) getString(R.string.search_namespaces) else namespaceMap[callback?.getLangCode()]?.get(ns).orEmpty() + ":"
+            itemView.isEnabled = ns != null
+            itemView.setTextColor(ResourceUtil.getThemedColor(requireContext(), if (ns == null) R.attr.material_theme_primary_color else R.attr.colorAccent))
+        }
+
+        override fun onClick(v: View) {
+            callback?.switchToSearch((v as TextView).text.toString())
+        }
+    }
+
+    private inner class NamespaceAdapter : RecyclerView.Adapter<NamespaceItemViewHolder>() {
+        override fun getItemCount(): Int {
+            return namespaceHints.size + 1
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NamespaceItemViewHolder {
+            return NamespaceItemViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_namespace, parent, false))
+        }
+
+        override fun onBindViewHolder(holder: NamespaceItemViewHolder, pos: Int) {
+            holder.bindItem(if (pos > 0) namespaceHints[pos - 1] else null)
```

Comment:
We can use `namespaceHints.getOrNull(pos - 1)` here.

Reply:
yes!